File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-4.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        if not matrix:
            return None
        m = len(matrix)
        n = len(matrix[0])
        rows = set()
        columns = set()
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 0:
                    rows.add(i)
                    columns.add(j)

        for row in rows:
            for j in range(n):
                matrix[row][j] = 0
        
        for i in range(m):
            for column in columns:
                matrix[i][column] = 0
                

File: Data Structures & Algorithms/search-2d-matrix/submission-4.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        for(auto it: matrix){
            int L = 0;
            int R = it.size() - 1;
            while(L <= R){
                int MID = (L + R)/2;
                if(it[MID] == target){
                    return true;
                }
                else if(target > it[MID]){
                    L = MID + 1;
                }
                else{
                    R = MID - 1;
                }
            }
        }
        return false;
    }
};
